Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java

```diff
@@ -0,0 +1,86 @@
+package com.novoda.bonfire.channel.view;
+
+import android.content.Context;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.Switch;
+
+import com.novoda.bonfire.R;
+import com.novoda.bonfire.channel.displayer.NewChannelDisplayer;
+import com.novoda.notils.caster.Views;
+
+public class NewChannelView extends LinearLayout implements NewChannelDisplayer {
+
+    private ChannelCreationListener channelCreationListener = ChannelCreationListener.NO_OP;
+    private EditText newChannelName;
+    private Switch privateChannelSwitch;
+    private Button createButton;
+
+    public NewChannelView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        setOrientation(VERTICAL);
+    }
+
+    @Override
+    protected void onFinishInflate() {
+        super.onFinishInflate();
+        View.inflate(getContext(), R.layout.merge_create_channel_view, this);
+        newChannelName = Views.findById(this, R.id.newChannelName);
+        privateChannelSwitch = Views.findById(this, R.id.privateChannel);
+        createButton = Views.findById(this, R.id.createButton);
+    }
+
+    @Override
+    public void attach(final ChannelCreationListener channelCreationListener) {
+        this.channelCreationListener = channelCreationListener;
+        newChannelName.addTextChangedListener(channelNameTextWatcher);
+        createButton.setOnClickListener(createButtonClickListener);
+        createButton.setEnabled(false);
+    }
+
+    @Override
+    public void detach(ChannelCreationListener channelCreationListener) {
+        this.channelCreationListener = ChannelCreationListener.NO_OP;
```

Comment:
@Dorvaryn it should be removed, the idea was to have an empty implementation of the listener so that it can never be null, but I guess that just hides a problem of calling stuff that shouldn't be called when the presenter is not attached